toolchain/sem_ir/inst.h

```diff
@@ -493,6 +493,15 @@ class InstStore {
     return GetCanonicalLocId(GetNonCanonicalLocId(inst_id));
   }
 
+  // Works like `GetCanonicalLocId`, but called as part of desugaring. Marks the
+  // returned location as desugared.
+  auto GetDesugaredLocId(LocId loc_id) const -> LocId {
```

Comment:
I'm finding the naming here a bit confusing. I think the issue is that I think of desugaring as a process of going from a wrapped / sugared thing and removing the wrapping to produce the underlying desugared thing, but what we're actually doing *to the location* here is more like the opposite of that: we're starting from a location in the source, and effectively *wrapping* it in something that says it's the location of the desugared code rather than the original location.

I wonder if there's some small change we can make here so that this reads as "get the location for the desugaring of the instruction at this location" rather than "get the desugared version of this location". Maybe something like:
```suggestion
  // Returns a virtual location to use for the desugaring of the code at the
  // specified location.
  auto GetLocForDesugaringOf(LocId loc_id) const -> LocId {
```

Reply:
I find the "Of" hard to place, does just `GetLocIdForDesugaring` work for you?

Reply:
Sure, that works for me too.

Reply:
I'd also be happy with dropping the `Id` if you want to shorten it a bit more.

Reply:
I like having `LocId` in these to emphasize the type involved; it's also parallel with `GetWithLocId`, `SetLocId`, `GetCanonicalLocId`, etc